src/aiida/orm/entities.py

```diff
@@ -99,23 +99,27 @@ def query(
         self,
         filters: Optional['FilterType'] = None,
         order_by: Optional['OrderByType'] = None,
+        project: Optional[Union[list[str], str]] = None,
         limit: Optional[int] = None,
         offset: Optional[int] = None,
+        subclassing: bool = True,
     ) -> 'QueryBuilder':
         """Get a query builder for the objects of this collection.
 
         :param filters: the keyword value pair filters to match
         :param order_by: a list of (key, direction) pairs specifying the sort order
+        :param project: Optional projections.
         :param limit: the maximum number of results to return
         :param offset: number of initial results to be skipped
+        :param subclassing: whether to match subclasses of the type as well.
         """
         from . import querybuilder
 
         filters = filters or {}
         order_by = {self.entity_type: order_by} if order_by else {}
 
         query = querybuilder.QueryBuilder(backend=self._backend, limit=limit, offset=offset)
-        query.append(self.entity_type, project='*', filters=filters)
+        query.append(self.entity_type, project=project, filters=filters, subclassing=subclassing)
```

Comment:
I guess this will change the initial behavior? Originally it will project on all entity_type, but with new `project` argument if it is not passed, it will project nothing?

Reply:
It won't because `project=None` is equivalent to `project='*'`:
```python
In [1]: QueryBuilder(limit=2).append(Node).all(flat=True)
Out[2]: 
[<CalcJobNode: uuid: c4b07b1e-7a99-458c-a017-e289b5d07ebb (pk: 90126) (aiida.calculations:core.shell)>,
 <UpfData: uuid: 7825b3b0-add8-4e09-ad06-804d197ae975 (pk: 115724)>]

In [2]: QueryBuilder(limit=2).append(Node, project=None).all(flat=True)
Out[2]: 
[<CalcJobNode: uuid: c4b07b1e-7a99-458c-a017-e289b5d07ebb (pk: 90126) (aiida.calculations:core.shell)>,
 <UpfData: uuid: 7825b3b0-add8-4e09-ad06-804d197ae975 (pk: 115724)>]
```

Reply:
Cool, thanks! Then all fine with this.